worker/peergrouper/desired.go

```diff
@@ -173,40 +192,45 @@ func desiredPeerGroup(info *peerGroupInfo) (bool, map[string]*replicaset.Member,
 	//    are not eligible to be primary.
 	// 3) Remove them.
 	// 4) Do nothing.
-	err := peerChanges.checkExtraMembers(info.extra)
+	err := p.checkExtraMembers()
 	if err != nil {
-		return false, nil, nil, errors.Trace(err)
+		return desiredChanges{}, errors.Trace(err)
 	}
 
-	peerChanges.members = info.initNewReplicaSet()
-	peerChanges.possiblePeerGroupChanges(info)
-	peerChanges.reviewPeerGroupChanges(info)
-	peerChanges.createNonVotingMember(&info.maxMemberId)
+	p.desired.members = p.info.initNewReplicaSet()
+	p.possiblePeerGroupChanges()
+	p.reviewPeerGroupChanges()
+	p.createNonVotingMember()
 
 	// Set up initial record of machine votes. Any changes after
 	// this will trigger a peer group election.
-	peerChanges.getMachinesVoting()
-	peerChanges.adjustVotes()
+	p.getMachinesVoting()
+	p.adjustVotes()
 
-	if err := peerChanges.updateAddresses(info); err != nil {
-		return false, nil, nil, errors.Trace(err)
+	if err := p.updateAddresses(p.info); err != nil {
```

Comment:
I would follow suite with the other changes; remove the argument and get it from the receiver.

Reply:
yeah, I missed this one, just noticed myself.